cms/templates/studio_xblock_wrapper.html

```diff
@@ -104,9 +104,17 @@
                                 <div class="nav-sub">
                                   <ul>
                                     % if not show_inline:
-                                        <li class="nav-item">
-                                            <a class="copy-button" href="#" role="button">${_("Copy to Clipboard")}</a>
-                                        </li>
+                                        % if is_course and can_add:
+                                            <!--
+                                                Only show the "Copy to Clipboard" button for xblocks inside courses where
+                                                we can add. This prevents it from showing in ContentLibrary page and in
+                                                ContentLibrary preview from within a course unit's component. As the copy/paste
```

Comment:
> and in ContentLibrary preview from within a course unit's component

Actually, why do we need to hide it from there? Technically, the preview within a course is just part of the course, so can we actually allow copying from there? Or is it not working for some reason?

Reply:
@bradenmacdonald This is the preview of the contents of the LibraryContent inside the course. The one you access by clicking on the View button of the component in the Unit:

![Screen Shot 2023-09-21 at 12 39 01 PM](https://github.com/openedx/edx-platform/assets/6829768/ce2e6490-4801-4ac8-98ec-03d55816ebfa)

I thought this was sort of like a read-only view of the LibraryContent block you have in your unit, and any changes you want to make you'd need to make them in the actual Library. I noticed there were none of the other actions there as well, like delete/duplicate, and when I test Copy to Clipboard it was broken (though due to the paste button/preview not being set up properly). So because of that I thought the Copy to Clipboard functionality was not intended to be there? If that is not the case, I can look into fixing the functionality up on that page and bringing it back.

Reply:
@yusuf-musleh Yes, that's the page that I thought it was. Technically the components you see there are actually editable copies of the library content - you can Edit the individual components and override some of the fields if you want to (like how many attempts on a problem are allowed).

We definitely don't want to allow _pasting_ on those pages, but in principle there's no reason we can't allow copying - if it works. I hadn't actually thought about whether or not copying should be on those pages until I saw your PR, but I think if there's some very simple reason it wasn't working, we should fix the bug so it works rather than disable the option.

However, if it seems like it will take a lot of time to fix the issue and make copying work on those blocks, we can stick with your approach here of hiding copy. It's not a big deal either way.

Reply:
@bradenmacdonald Got it. Since we didn't want to allow the _pasting_ on those pages, fixing the _copy_ functionality was pretty straightforward. I just prevented the `refreshPasteButton` functionality from running when in those pages since the paste button is not initialized there. Copying seems to be working fine now, I managed to copy component and paste it in another unit with no issues.